src/main/scala/uk/gov/hmrc/ct/ct600/v3/retriever/CT600BoxRetriever.scala

```diff
@@ -16,67 +16,64 @@
 
 package uk.gov.hmrc.ct.ct600.v3.retriever
 
-import uk.gov.hmrc.ct.accounts.retriever.AccountsBoxRetriever
-import uk.gov.hmrc.ct.box.retriever.FilingAttributesBoxValueRetriever
+import uk.gov.hmrc.ct.box.retriever.BoxRetriever
 import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
 import uk.gov.hmrc.ct.ct600.v3._
 import uk.gov.hmrc.ct.ct600a.v3.retriever.CT600ABoxRetriever
 
-trait CT600BoxRetriever extends ComputationsBoxRetriever with CT600DeclarationBoxRetriever with AboutThisReturnBoxRetriever {
-
-  self: AccountsBoxRetriever with FilingAttributesBoxValueRetriever =>
+abstract class CT600BoxRetriever(computationsBoxRetriever: ComputationsBoxRetriever) extends BoxRetriever {
 
   def b1(): B1
 
-  def b2(): B2 = B2(ac1())
+  def b2(): B2 = B2(computationsBoxRetriever.accountsBoxRetriever.ac1())
 
-  def b3(): B3 = B3(utr())
+  def b3(): B3 = B3(computationsBoxRetriever.filingAttributesBoxValueRetriever.utr())
```

Comment:
As in Tomasz's comment above, not sure about this though. Do you think each retriever should have the retrievers it needs as a first level val instead of chaining it? I.e. A CT600 will always need comps, accounts and filing. 
I'm not sure and am happy to merge this, just worth thinking about.

Reply:
Agreed. It will make the concrete retrievers a little more verbose (IE have more things injected) but is probably nicer.

I'll see what impact it has.